### ISSUE TITLE
Tenant label for non static configs

### ISSUE THREAD

**Issue by scotforescout** [blind-oracle/cortex-tenant#5]
In the [Prometheus Example Config](https://github.com/blind-oracle/cortex-tenant#prometheus-configuration-example) there are only references for adding in the tenant label for static configs.

How does one go to add them for other jobs? Would something like this work for all services from a prometheus server?

```
global:
  external_labels:
    tenant: foobar
```

Thanks for making this btw, very clean.

**Comment by blind-oracle** [comment]
@scotforescout Yes, `external_labels` should work just fine - it will add the specified label to all jobs. Or, if you want per-job labels you can use `relabel_config` to add label like explained here: https://albersdevelopment.net/2019/08/28/prometheus-adding-a-label-to-a-target/

**Comment by scotforescout** [comment]
Perfect thanks, I'm triaging an issue while turning on the authentication my dev cortex cluster. As soon as I turned it on, there's a ton of out of order errors that actually choked the cluster (while auth_enabled: false it doesn't have the issue) but it looks mostly like an issue on how prometheus servers are scraping service discovery endpoints.

**Comment by blind-oracle** [comment]
Out-of-order thing is usually caused by scraping some federated endpoint which has some time skew problems. Try setting `honor_timestamps` to false and see if that helps. This way Prometheus will override metrics timestamps using local time.

Also if you're using HA Cortex setup that might be the cause.

**Comment by scotforescout** [comment]
I actually found the problem and I'll try my best to describe what I was going through.

Its all stems from this [issue](https://github.com/cortexproject/cortex/issues/2662) which needs to probably get highlighted as a major blocker for people trying to implement a cortex cluster.

This wasn't a problem until after I enabled authentication and a tenant ID from this service, so it felt like tenant IDs and cortex-tenant was the source of the issue, but it wasn't.

A prom server that does relabelling & metric relabelling but doesn't put enough identifying labels for a remote write will cause a cortex server to choke. 

My prometheus configuration was taking a job's labels, removing the majority of the labels and a bunch of them as metric labels. It only had 2 identifying labels, IP & Port. (This was done with the cAdvisor metrics which at the time didn't have a good way to remove labels through the service and had a bunch of extra garbage metrics I didn't want to process)

So what happened is that cortex-tenant adds in another label and then cortex tried to reprocess the metric, but it found similar metrics because I removed unique labels to distinguish them from each other. 

It didn't have this problem when I had `auth_enabled: false` because I believe it was just a straight copy of the metric, it didn't have to add in an additional label with the tenant.

This is all observations from just a day playing around with it.
